src/tox_current_env/hooks4.py

```diff
@@ -0,0 +1,241 @@
+        env_conf.loaders.insert(0, allow_external_cmds)
+    # For print-deps-to and print-extras-to, use empty
+    # list of commands so the tox does nothing.
+    if opt.print_deps_to or opt.print_extras_to:
+        empty_commands = MemoryLoader(commands=[], commands_pre=[], commands_post=[])
+        env_conf.loaders.insert(0, empty_commands)
+
+
+class Installer:
+    """Noop installer"""
+
+    def install(self, *args, **kwargs):
+        return None
+
+
+class CurrentEnvLocalSubProcessExecutor(Execute):
+    def build_instance(
+        self,
+        request,
+        options,
+        out,
+        err,
+    ):
+        request.env["PATH"] = ":".join(
+            (str(options._env.env_dir / "bin"), request.env.get("PATH", ""))
+        )
+        return LocalSubProcessExecuteInstance(request, options, out, err)
+
+
+class CurrentEnv(PythonRun):
+    def __init__(self, create_args):
+        self._executor = None
+        self._installer = None
+        self._path = []
+        super().__init__(create_args)
+
+    @staticmethod
+    def id():
+        return "current-env"
+
+    @property
+    def _default_package_tox_env_type(self):
+        return None
+
+    @property
+    def _external_pkg_tox_env_type(self):
+        return None
+
+    @property
+    def _package_tox_env_type(self):
+        return None
+
+    @property
+    def executor(self):
+        if self._executor is None:
+            self._executor = CurrentEnvLocalSubProcessExecutor(self.options.is_colored)
+        return self._executor
+
+    def _get_python(self, base_python):
+        return PythonInfo(
+            implementation=sys.implementation,
+            version_info=sys.version_info,
+            version=sys.version,
+            is_64=(platform.architecture()[0] == "64bit"),
+            platform=platform.platform(),
+            extra={"executable": Path(sys.executable)},
+        )
+
+    def create_python_env(self):
+        """Fake Python environment just to make sure all possible
+        commands like python or python3 works."""
+        bindir = self.env_dir / "bin"
+        if not bindir.exists():
+            os.mkdir(bindir)
+        for suffix in (
+            "",
+            f"{sys.version_info.major}",
+            f"{sys.version_info.major}.{sys.version_info.minor}",
+        ):
+            symlink = bindir / f"python{suffix}"
+            if not symlink.exists():
+                os.symlink(sys.executable, symlink)
+
+    def env_bin_dir(self):
+        return Path(sysconfig.get_path("scripts"))
+
+    def env_python(self):
+        return sys.executable
+
+    def env_site_package_dir(self):
+        return Path(sysconfig.get_path("purelib"))
+
+    @property
+    def installer(self):
+        return Installer()
+
+    def prepend_env_var_path(self):
+        return [self.env_bin_dir()]
+
+    @property
+    def runs_on_platform(self):
+        return sys.platform
+
+
+class PrintEnv(CurrentEnv):
```

Comment:
What is this trying to achieve?

Reply:
The environment or the inheritance?

Reply:
This entire class 🤔 not sure what's its goal is.

Reply:
The goal of this class is to provide an implementation for options that print information about deps and extras rather thane execute tests.

Reply:
Perhaps it would be better served by a `deps` sub-command 😊 as part of the core.

Reply:
Perhaps we could deprecate this option then, but for now we need to maintain compatible behavior between the tox plugin for tox 3 and 4.